src/lib.rs

```diff
@@ -62,9 +62,25 @@ pub enum ClangFormatError {
 }
 
 /// Execute clang-format with the given input, using the given style, and collect the output
-fn clang_format_with_style(
+///
+/// # Example
+///
+/// ```
+/// # use clang_format::{clang_format_with_style, ClangFormatStyle};
+/// # fn main() {
+/// let input = r#"
+///     struct Test {
+///
+///     };
+/// "#;
+/// let output = clang_format_with_style(input, ClangFormatStyle::Mozilla);
+/// assert!(output.is_ok());
+/// assert_eq!(output.unwrap(), "\nstruct Test\n{};\n");
+/// # }
+/// ```
+pub fn clang_format_with_style(
     input: &str,
-    style: &ClangFormatStyle,
+    style: ClangFormatStyle,
```

Comment:
why did this change to consuming the style?

Reply:
When making the `clang_format_with_style` public i wondered which of the following felt more natural ?

Maybe be-ref is fine though, i can change it back :-) 

```rust
clang_format_with_style(input, ClangFormatStyle::Mozilla);
or
clang_format_with_style(input, &ClangFormatStyle::Mozilla);
```

Reply:
Reverted back to by-ref